snapshots/overlay/overlay_test.go

```diff
@@ -329,34 +345,105 @@ func testOverlayView(t *testing.T, newSnapshotter testsuite.SnapshotterFunc) {
 	}
 
 	supportsIndex := supportsIndex()
-	expectedOptions := 3
+	expectedOptions := 4
 	if !supportsIndex {
 		expectedOptions--
 	}
-	userxattr, err := overlayutils.NeedsUserXAttr(root)
-	if err != nil {
-		t.Fatal(err)
-	}
-	if userxattr {
-		expectedOptions++
-	}
 
 	if len(m.Options) != expectedOptions {
 		t.Errorf("expected %d additional mount option but got %d", expectedOptions, len(m.Options))
 	}
 	lowers := getParents(ctx, o, root, "/tmp/view2")
 	expected = fmt.Sprintf("lowerdir=%s:%s", lowers[0], lowers[1])
-	optIdx := 2
+	optIdx := 3
 	if !supportsIndex {
 		optIdx--
 	}
-	if userxattr {
-		optIdx++
-	}
 	if m.Options[0] != "volatile" {
 		t.Error("expected option first option to be provided option \"volatile\"")
 	}
 	if m.Options[optIdx] != expected {
 		t.Errorf("expected option %q but received %q", expected, m.Options[optIdx])
 	}
 }
+
+func testOverlayMetacopy(t *testing.T, newSnapshotter testsuite.SnapshotterFunc, metacopyOn bool) {
+	ctx := context.TODO()
+	root := t.TempDir()
+	o, _, err := newSnapshotter(ctx, root)
+	if err != nil {
+		t.Fatal(err)
+	}
+	key := "/tmp/base"
+	// bind mount -- doen't have metacopy option
+	mounts, err := o.Prepare(ctx, key, "")
+	if err != nil {
+		t.Fatal(err)
+	}
+	m := mounts[0]
+	if m.Type != "bind" {
+		t.Fatal("mount without parents must be a \"bind\" mount")
+	}
+	if len(m.Options) != 2 {
+		t.Fatalf("too many mount options - %d, expected 2", len(m.Options))
+	}
+	err = o.Commit(ctx, "base", key)
+	if err != nil {
+		t.Fatal(err)
+	}
+	key = "/tmp/top"
+	// active snapshot -- may or may not have metacopy option
+	mounts, err = o.Prepare(ctx, key, "base")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var (
+		bp    = getBasePath(ctx, o, root, key)
+		work  = "workdir=" + filepath.Join(bp, "work")
+		upper = "upperdir=" + filepath.Join(bp, "fs")
+		lower = "lowerdir=" + getParents(ctx, o, root, key)[0]
+	)
+	expected := []string{
+		"index=off",
+	}
+	if !supportsIndex() {
+		expected = expected[1:]
+	}
+	userxattr, err := overlayutils.NeedsUserXAttr(root)
+	if err != nil {
+		t.Fatal(err)
+	} else if userxattr {
+		expected = append(expected, "userxattr")
```

Comment:
This looks wrong to me, actually `testOverlayOverlayMount` does the same, but according to the sequence in `snapshots/overlay/overlay.go` if `userxattr` and `index` is supported then we should have `m.Options = ["userxattr", "index=off"]`, but `expected = ["index=off", "userxattr"]`?